### PR TITLE
Fix wrong getHeader refactoring

### PR DESCRIPTION
**Pull request by Alkarex**
Fix regression introduced by https://github.com/FreshRSS/FreshRSS/pull/2373
The refactoring required a call to init() even for static functions, which is most of the time not done.
Removed premature abstraction of `$_SERVER`, which was the root cause of the bug.
https://github.com/FreshRSS/FreshRSS/issues/2748#issuecomment-569898931

### CODE REVIEW

File: lib/Minz/Request.php [hunk 1]
```diff
@@ -227,6 +225,42 @@ public static function getBaseUrl() {
...
+	 *
+	 * Note: for the moment it tests only if address is corresponding to a
+	 * localhost address.
+	 *
+	 * @param $address the address to test, can be an IP or a URL.
+	 * @return true if server is accessible, false otherwise.
+	 * @todo improve test with a more valid technique (e.g. test with an external server?)
+	 */
+	public static function serverIsPublic($address) {
+		if (strlen($address) < strlen('http://a.bc')) {
```
Comment (Frenzie) [inline lib/Minz/Request.php:239]: We're sure it will always includes a protocol at this point?

Comment (Alkarex) [reply lib/Minz/Request.php:239]: If it does not, then it is not a valid public address

**Review by aledeg — COMMENTED** [review]
I am not a big fan of the method used to check the url. Can't you use `filter_var` with a built-in filter instead?

- [PHP: Validate filters - Manual](https://www.php.net/manual/en/filter.filters.validate.php)
- [PHP: filter_var - Manual](https://www.php.net/manual/en/function.filter-var.php)

**Comment by Alkarex** [comment]
@aledeg Which line(s) more precisely?

**Comment by aledeg** [comment]
This part
```
if (strlen($address) < strlen('http://a.bc')
```

and also the one with the `preg_match`

I am not sure if it's doable though

**Comment by Alkarex** [comment]
Sorry for the delay. I will try to look at it tomorrow, but I cannot immediately think of a better way.
The one with `strlen($address)` is just to relax a previous `== ''` now that we might have an empty string or something like `http:`

**Comment by Alkarex** [comment]
@aledeg I do agree that we should improve the way we guess and/or test whether an URL is public/routable. Suggestions welcome. However, I am not sure how to do that better at the moment, and this PR is urgent, as all current installations based on /master are broken

**Comment by Alkarex** [comment]
Sorry to be a bit pushy, but this is quite an urgent bug. So let me merge and let's continue improvements in another PR if needed

**Pull request merged**